packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1692,11 +1712,21 @@ function applyVirtualResolutionMutations({
   const virtualStack = new Map<LocatorHash, number>();
   const resolutionStack: Array<Locator> = [];
 
+  const allIdents = new Map<IdentHash, Ident>();
+
   // We'll be keeping track of all virtual descriptors; once they have all
   // been generated we'll check whether they can be consolidated into one.
   const allVirtualInstances = new Map<LocatorHash, Map<string, Descriptor>>();
   const allVirtualDependents = new Map<DescriptorHash, Set<LocatorHash>>();
 
+  // First key is the first package that provides the peer dependency. Second
```

Comment:
Isn't it rather the first package that peer depends on the peer dependency? If I understand the code correctly, the parent of this package is the one actually providing the peer dependency 🤔

Reply:
Correct 👍